### PR TITLE
Messaging feature

### PR DESCRIPTION
**Pull request by Jacobjeevan**
Apologize for all the formatting differences. I added prettier (for linting) to packages, so shouldn't happen again.

- wishCardFullPage.ejs
	- Dropdown -> select
	- Allows user to select a message and send it
	- Displays existing messages, along with User it's from

- wishCardFullPage.js
	- Call to backend for submitting the message
	- To Do: prettify error messages/alerts

- wishCards.js route
	- Modified wishcard get (by id) method to populate messages and as well user.
	- Slight modifications to post /message for updating the wishcards.

Additional comments:

For a guest, clicking on view more on wish card page leads to login page. Alternatively, we could display an alert instead (or maybe do a combination of both).

For now the population approach works, but it might not scale well (performance wise, especially if we have a lot of cards). But, we can worry about that later :D

- Changed AWS bucket name to refer to config file

### CODE REVIEW

File: routes/wishCards.js [hunk 1]
```diff
@@ -10,201 +10,259 @@ const router = express.Router();
...
-        res.status(400).send(JSON.stringify({
-            success: false,
-            error: error
-        }));
-    }
+  try {
+    const results = await WishCard.find({
+      wishItemName: {
+        $regex: req.body.wishitem,
+        $options: 'i',
```
Comment (gnorbsl) [inline routes/wishCards.js:127]: double ticks got removed here is that an issue?

File: views/wishCardFullPage.ejs [hunk 2]
```diff
@@ -8,6 +8,10 @@
     <!-- ** NOTE: CDN Bootstrap.js, Popper.js, and jQuery are moved to the bottom so the page loads faster-->
     <%- include('./templates/header')-%>
     <link rel="stylesheet" href="../css/wishCardFull.css">
+    <!-- jquery -->
+    <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.5.1/jquery.min.js"></script>
+    <!-- js scripts -->
+    <script src="../clientJS/wishCardFullPage.js"></script>
```
Comment (gnorbsl) [inline views/wishCardFullPage.ejs:14]: we dont need double dots, /clientJS/ should be enough and wont cause issues in nested routes, double dots will.

File: views/wishCardFullPage.ejs [hunk 3]
```diff
@@ -40,44 +44,39 @@
...
+                        <option value="7"><%= wishcard.childFirstName %>, you are awesome!</option>
+                        <option value="8">Lots of love and best wishes, <%= wishcard.childFirstName %></option>
+                        <option value="9"><%= wishcard.childFirstName %>, hope you enjoy my gift!</option>
+                        <option value="10">Merry Christmas and a Happy New Year</option>
+                        <option value="11"><%= wishcard.childFirstName %>, have a happy holiday</option>
+                        <option value="12">Congratulations, <%= wishcard.childFirstName %></option>
+                    </select>
+                    <input type="submit" id="submit" class="mt-4" value="Send Message">
+                </form>
+
```
Comment (gnorbsl) [inline views/wishCardFullPage.ejs:66]: I would suggest to generate an array in the backend and just iterate over it here, gives us more flexibilty and cleans up the code. maybe we can add a placeholder in the strings like %name% and replace every occurence of the placeholder with the name.

Not neccessary though

File: views/wishCardFullPage.ejs [hunk 4]
```diff
@@ -40,44 +44,39 @@
...
+                </form>
+                
             </div>
 
             <!-- This part needs to show all users' messages sent to this wishcard 
             wishcard model has 'messages' field which is an objectId array of 'Message'-->
-            <div class="show-msg"> 
-                <% if (wishcard.messages.length === 0) { %>
+            <div class="show-msg">
+                <% if (messages.length === 0) { %>
```
Comment (gnorbsl) [inline views/wishCardFullPage.ejs:72]: miising wishcard.?

Comment (gnorbsl) [reply views/wishCardFullPage.ejs:72]: Ah different variables, a bit confusing, whats the difference here between wishcard.messages and messages, maybe a different variable name to clarify?

File: views/wishCardFullPage.ejs [hunk 5]
```diff
@@ -40,44 +44,39 @@
...
                 <h6 class="comf-font">No one signed <%= wishcard.childFirstName %>'s card yet.
                     Be the first person to send a message.</h6>
                 <% } else { %>
                 <div>
-                    <% for (var i = 0; i < wishcard.messages.length; i++) { %>
-                    <p>This line needs to show each user's message sent to this wishcard</p>
-                    <p>Sent by: <%= user.fName %></p>
+                    <% for (var i = 0; i < messages.length; i++) { %>
```
Comment (gnorbsl) [inline views/wishCardFullPage.ejs:77]: I suggest using let instead of var since var is kind of deprecated

**Review by gnorbsl — CHANGES_REQUESTED** [review]

File: views/wishCardFullPage.ejs (hunk 4 above)
Comment (Jacobjeevan) [reply views/wishCardFullPage.ejs:72]: Yeah, probably a good idea to do that. I will change 73 to just messages.

File: routes/wishCards.js [hunk 6]
```diff
@@ -5,206 +5,264 @@ serving all wishcard related routes
...
-        cb(null, false);
-    }
-}
+  // reject a file
+  if (
+    file.mimetype === "image/jpeg" ||
+    file.mimetype === "image/jpg" ||
+    file.mimetype === "image/gif" ||
+    file.mimetype === "image/png"
+  ) {
```
[the remark this answers is not in the document]
Comment (Jacobjeevan) [reply routes/wishCards.js:46]: Lmao

File: routes/wishCards.js (hunk 1 above)
Comment (Jacobjeevan) [reply routes/wishCards.js:127]: Nah, double ticks aren't necessary there :+1:

File: views/wishCardFullPage.ejs (hunk 5 above)
Comment (stacysealky) [reply views/wishCardFullPage.ejs:77]: sorry that was me lol too used to using var

File: views/wishCardFullPage.ejs (hunk 4 above)
Comment (gnorbsl) [reply views/wishCardFullPage.ejs:72]: ah wait you replaced all wishcard.messages with messages?
sorry git is a bit confusing sometimes

**Comment by gnorbsl** [comment]
@Jacobjeevan can you change the name of the bucket var into S3BUCKET with uppercase?

File: views/wishCardFullPage.ejs (hunk 4 above)
Comment (Jacobjeevan) [reply views/wishCardFullPage.ejs:72]: Yeah, no worries :+1:

**Pull request merged**